### ISSUE TITLE
net: Conn.Close doesn't unblock Conn.Read on nacl 

### ISSUE THREAD

**Issue by bradfitz** [golang/go#17695]
It appears that on nacl, a net.Conn.Close (on a "TCP" localhost conn)  doesn't unblock a pending Conn.Read.

I thought we had a test for this and it worked, but I'm probably remembering a similar case instead.

Add a test & fix.

Failing higher-level test (skipped for now) is in net/http's TestServerSetKeepAlivesEnabledClosesConns.

Low priority (nacl).

**Comment by rsc** [comment]
Good for freeze I guess. That code should be going into syscall/net_nacl.go's netFile and then byteq.read and .close, and as far as I can tell, a close should be waking up a read.
